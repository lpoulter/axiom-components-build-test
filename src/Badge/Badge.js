import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './Badge.css';

export default class Badge extends Component {
  static propTypes = {
    /** Content to appear inside the Badge */
    children: PropTypes.node,
    /** Colour of the Badge */
    color: PropTypes.oneOf([
      'faded',
      'highlight',
      'success',
      'error',
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
    ]).isRequired,
    /**
     * Controls the full width appearance of the badge either all of the time,
     * with a value of `true` otherwise at one of the breakpoints specified.
     */
    full: PropTypes.oneOf([true, 'small', 'medium', 'large']),
    /** Click handler that applied clickable styling */
    onClick: PropTypes.func,
    /** The opacity of the background of the Badge */
    opacity: PropTypes.number,
  };

  static defaultProps = {
    color: 'faded',
    opacity: 1,
  };

  render() {
    const {
      children,
      color,
      full,
      onClick,
      opacity: o,
      ...rest
    } = this.props;

    const isFaded = color === 'faded';
    const textColor = isFaded || o < 1 ? undefined : 'night';
    const opacity = isFaded || o < 1 ? Math.max(.1, Math.min(isFaded ? .3 : .6, o)) : 1;
    const style = { opacity };
    const classes = classnames('ax-badge', {
      [`ax-badge--${color}`]: color,
      'ax-badge--clickable': onClick,
      'ax-badge--full': full === true,
      [`ax-badge--full--${full}`]: typeof full === 'string',
    });

    return (
      <Base { ...rest } Component="span" className={ classes } onClick={ onClick } textColor={ textColor }>
        <span className="ax-badge__background" style={ style } />
        <span className="ax-badge__content">{ children }</span>
      </Base>
    );
  }
}
