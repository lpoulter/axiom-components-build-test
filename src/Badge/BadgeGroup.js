import PropTypes from 'prop-types';
import React, { Component } from 'react';
import InlineGroup from '../InlineGroup/InlineGroup';

export default class BadgeGroup extends Component {
  static propTypes = {
    /** Badges inside the group */
    children: PropTypes.node,
  };

  render() {
    const { children, ...rest } = this.props;

    return (
      <InlineGroup space="x4" { ...rest } className="ax-badge-group">
        { children }
      </InlineGroup>
    );
  }
}
