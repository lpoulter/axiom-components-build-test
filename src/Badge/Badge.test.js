import React from 'react';
import renderer from 'react-test-renderer';
import Badge from './Badge';

const getComponent = (props = {}) =>
  renderer.create(
    <Badge { ...props }>
      Lorem Ipsum
    </Badge>
  );

describe('Badge', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders opacity minimum 0.1', () => {
    const component = getComponent({ opacity: 0, color: 'success' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders opacity maximum 0.6', () => {
    const component = getComponent({ opacity: 0.7, color: 'success' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders opacity minimum 0.1 with color faded', () => {
    const component = getComponent({ opacity: 0, color: 'faded' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders opacity maximum 0.3 with color faded', () => {
    const component = getComponent({ opacity: 0.7, color: 'faded' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders opacity 1', () => {
    const component = getComponent({ opacity: 1, color: 'success' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does not render opacity 1 with color faded', () => {
    const component = getComponent({ opacity: 1, color: 'faded' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders onClick', () => {
    const component = getComponent({ onClick: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders opacity', () => {
    const component = getComponent({ opacity: 0.5, color: 'success' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders opacity with color faded', () => {
    const component = getComponent({ opacity: 0.2, color: 'faded' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  describe('renders with color', () => {
    [
      'faded',
      'highlight',
      'success',
      'error',
      'tiny-clanger',
      'critical-mass',
      'paradise-lost',
      'serene-sea',
      'giant-leap',
      'moon-lagoon',
      'terra-form',
      'primeval-soup',
      'new-horizon',
      'blast-off',
      'ground-control',
      'luna-dust',
    ].forEach((color) => {
      it(color, () => {
        const component = getComponent({ color });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with full', () => {
    [true, 'small', 'medium', 'large'].forEach((full) => {
      it(String(full), () => {
        const component = getComponent({ full });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
