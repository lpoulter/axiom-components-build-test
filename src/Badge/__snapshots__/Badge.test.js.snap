// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge does not render opacity 1 with color faded 1`] = `
<span
  className="ax-badge ax-badge--faded"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders onClick 1`] = `
<span
  className="ax-badge ax-badge--faded ax-badge--clickable"
  onClick={[Function]}
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders opacity 1`] = `
<span
  className="ax-badge ax-badge--success"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.5,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders opacity 1 1`] = `
<span
  className="ax-badge ax-badge--success ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders opacity maximum 0.3 with color faded 1`] = `
<span
  className="ax-badge ax-badge--faded"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders opacity maximum 0.6 1`] = `
<span
  className="ax-badge ax-badge--success"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.6,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders opacity minimum 0.1 1`] = `
<span
  className="ax-badge ax-badge--success"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders opacity minimum 0.1 with color faded 1`] = `
<span
  className="ax-badge ax-badge--faded"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders opacity with color faded 1`] = `
<span
  className="ax-badge ax-badge--faded"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.2,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color blast-off 1`] = `
<span
  className="ax-badge ax-badge--blast-off ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color critical-mass 1`] = `
<span
  className="ax-badge ax-badge--critical-mass ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color error 1`] = `
<span
  className="ax-badge ax-badge--error ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color faded 1`] = `
<span
  className="ax-badge ax-badge--faded"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color giant-leap 1`] = `
<span
  className="ax-badge ax-badge--giant-leap ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color ground-control 1`] = `
<span
  className="ax-badge ax-badge--ground-control ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color highlight 1`] = `
<span
  className="ax-badge ax-badge--highlight ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color luna-dust 1`] = `
<span
  className="ax-badge ax-badge--luna-dust ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color moon-lagoon 1`] = `
<span
  className="ax-badge ax-badge--moon-lagoon ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color new-horizon 1`] = `
<span
  className="ax-badge ax-badge--new-horizon ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color paradise-lost 1`] = `
<span
  className="ax-badge ax-badge--paradise-lost ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color primeval-soup 1`] = `
<span
  className="ax-badge ax-badge--primeval-soup ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color serene-sea 1`] = `
<span
  className="ax-badge ax-badge--serene-sea ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color success 1`] = `
<span
  className="ax-badge ax-badge--success ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color terra-form 1`] = `
<span
  className="ax-badge ax-badge--terra-form ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with color tiny-clanger 1`] = `
<span
  className="ax-badge ax-badge--tiny-clanger ax-text--color-night"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with defaultProps 1`] = `
<span
  className="ax-badge ax-badge--faded"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with full large 1`] = `
<span
  className="ax-badge ax-badge--faded ax-badge--full--large"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with full medium 1`] = `
<span
  className="ax-badge ax-badge--faded ax-badge--full--medium"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with full small 1`] = `
<span
  className="ax-badge ax-badge--faded ax-badge--full--small"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;

exports[`Badge renders with full true 1`] = `
<span
  className="ax-badge ax-badge--faded ax-badge--full"
>
  <span
    className="ax-badge__background"
    style={
      Object {
        "opacity": 0.3,
      }
    }
  />
  <span
    className="ax-badge__content"
  >
    Lorem Ipsum
  </span>
</span>
`;
